pandas/_libs/lib.pyx

```diff
@@ -1098,13 +1099,19 @@ def is_list_like(obj: object, allow_sets: bool = True) -> bool:
 
 
 cdef inline bint c_is_list_like(object obj, bint allow_sets) except -1:
+    # first, performance short-cuts for the most common cases
+    if cnp.PyArray_Check(obj):
+        return not cnp.PyArray_IsZeroDim(obj)
+    if PyList_Check(obj):
```

Comment:
im pretty sure cython will automatically optimize this from the regular isinstance check

Reply:
Are you sure? I was under the impression that `PyList_Check` does not check for subclasses and thus is probably faster. Anyway, `is_scalar` utilises C-API functions heavily in it's optimised short-cuts. I just followed that example. I can change that to `isinstance` if that is preferred.

Reply:
> Are you sure? 

from the cythonized internals.c:

```
  /* "pandas/_libs/internals.pyx":683
 *             return
 * 
 *         if isinstance(blocks, list):             # <<<<<<<<<<<<<<
 *             # Backward compat for e.g. pyarrow
 *             blocks = tuple(blocks)
 */
  __pyx_t_2 = PyList_Check(__pyx_v_blocks); 
```